src/target/riscv/riscv-013.c

```diff
@@ -916,16 +958,10 @@ static int register_write_abstract(struct target *target, enum gdb_regno number,
 	res = abstract_cmd_batch_check_and_clear_cmderr(target, batch,
 			abstractcs_read_key, &cmderr);
 
-	if (res != ERROR_OK) {
-		if (cmderr == CMDERR_NOT_SUPPORTED) {
-			if (number >= GDB_REGNO_FPR0 && number <= GDB_REGNO_FPR31) {
-				info->abstract_write_fpr_supported = false;
-				LOG_TARGET_INFO(target, "Disabling abstract command writes to FPRs.");
-			} else if (number >= GDB_REGNO_CSR0 && number <= GDB_REGNO_CSR4095) {
-				info->abstract_write_csr_supported = false;
-				LOG_TARGET_INFO(target, "Disabling abstract command writes to CSRs.");
-			}
-		}
+	if (res != ERROR_OK && cmderr == CMDERR_NOT_SUPPORTED) {
+		LOG_TARGET_DEBUG(target, "Caching the command as not supported");
```

Comment:
This print can contain the command number, same as above in `riscv013_execute_abstract_command()`:

```suggestion
		LOG_TARGET_DEBUG(target, "Caching the abstract command 0x% " PRIx32 " as not supported");
```

Even better, you can create small helper functions to mark/check if command is unsupported, and move the prints into these functions. That way, the prints will be located in a single place only:

```c
static void mark_command_as_unsupported(struct target* target, enum riscv_debug_reg_ordinal command_type, uint32_t command)
{
	LOG_TARGET_DEBUG(target, "Caching the command 0x% " PRIx32 " as not supported");
	log_debug_reg(target, command_type, command, __FILE__, __LINE__, __func__);
	RISCV013_INFO(info);
	ac_cache_insert(&info->ac_not_supported_cache, command);
}

static bool is_command_unsupported(struct target* target, uint32_t command)
{
	// similar
}
```

Reply:
Addressed